tests/pathfinding/test_fee_schedule.py

```diff
@@ -128,22 +120,143 @@ def test_fees_in_routing():
     tn.set_fee(2, 1)
     tn.set_fee(2, 3)
 
-    # Now let's try imbalance fees
-    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(200), FA(200))])
+    # Let's try imbalance fees
+
+    # Incoming channel
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(0)), (TA(200), FA(200))])
     assert tn.estimate_fee(1, 3) == 10
     assert tn.estimate_fee(3, 1) == -10
 
     # The opposite fee schedule should give opposite results
-    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(200)), (TA(200), FA(0))])
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(200)), (TA(200), FA(0))])
+    assert tn.estimate_fee(1, 3) == -10
+    assert tn.estimate_fee(3, 1) == 10
+
+    # Outgoing channel
+    tn.set_fee(2, 1)
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(200), FA(200))])
     assert tn.estimate_fee(1, 3) == -10
     assert tn.estimate_fee(3, 1) == 10
 
+    # The opposite fee schedule should give opposite results
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(200)), (TA(200), FA(0))])
+    assert tn.estimate_fee(1, 3) == 10
+    assert tn.estimate_fee(3, 1) == -10
+
+    # Combined fees cancel out
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(0)), (TA(200), FA(20))])
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(200), FA(20))])
+    assert tn.estimate_fee(1, 3) == 0
+    assert tn.estimate_fee(3, 1) == 0
+
+    tn.set_fee(2, 1, imbalance_penalty=[(TA(0), FA(20)), (TA(200), FA(0))])
+    tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(20)), (TA(200), FA(0))])
+    assert tn.estimate_fee(1, 3) == 0
+    assert tn.estimate_fee(3, 1) == 0
+
     # When the range covered by the imbalance_penalty does include the
     # necessary balance values, the route should be considered invalid.
     tn.set_fee(2, 3, imbalance_penalty=[(TA(0), FA(0)), (TA(80), FA(200))])
     assert tn.estimate_fee(1, 3) is None
 
 
+def test_fees_in_unbalanced_routing():
+    tn = TokenNetworkForTests(
+        channels=[
+            dict(participant1=1, participant2=2, capacity1=100, capacity2=0),
+            dict(participant1=2, participant2=3, capacity1=100, capacity2=0),
+        ]
+    )
+
+    # Make sure that routing works and the default fees are zero
+    assert tn.estimate_fee(1, 3) == 0
+
+    # Fees for the initiator are ignored
```

Comment:
Do we want to repeat these, or just focus on the imbalance fee in this test?

Reply:
I thought it would be good to have these as well, just to be sure. It's pretty fast to run...

Reply:
OK, I was just think aloud.